### ISSUE TITLE
Cookbook: using with pandas

### ISSUE THREAD

**Issue by onyxfish** [wireservice/agate#240]

**Comment by hunterowens** [comment]
Hey, is this still open? Would love to take a thwack at this sometime this week/weekend?

**Comment by onyxfish** [comment]
Yes, and please do! A couple things that I imagine could part of this:

* Tooling, process or documentation for converting agate tables to pandas tables
* As above, but vice-versa
* Examples of the sorts of analysis you can do in pandas and how that might feed back into what agate is good at

**Comment by onyxfish** [comment]
And for what it may be worth there is a half-functional effort to make a converter between agate and numpy [over here](https://github.com/onyxfish/agate-numpy).

**Comment by hunterowens** [comment]
Ok, I've added a Agate->Pandas method to the cookbook (see the WIP PR #337), but am wondering how to make an Agate table object from pandas. Any thoughts on useful methods? (ie, column inference?)

**Comment by onyxfish** [comment]
I haven't looked too closely at it. My understanding is that pandas dataframes are just numpy arrays under the hood, so I think a general solution to the numpy problem might work for both. Unfortunately, my efforts to reverse engineer numpys column structure have been frustrating. It's poorly documented and some information that you put into defining the column doesn't seem possible to get back out.

My hope, and intuition, was that pandas might use a subset of numpy's features that are easier to reverse. But that's only a hunch.

TL;DR, what I've done so far on [agate-numpy](https://github.com/onyxfish/agate-numpy) is all I've got at the moment.

**Comment by mathdesc** [comment]
Hello,

Reviving an old thread, still a _agate table_ to _panda table_ converter still looks today relevant.
There is a portage of [agate-lookup](https://github.com/wireservice/agate-lookup) to [panda-lookup ](https://github.com/ghing/pandas-lookup) exists and that looks promising toward that end.
Diffing `pandas-lookup/pandaslookup/` and `agate-lookup/agatelookup/`  reveals interesting things notably:

    for k, v in meta['columns'].items():
        converters[k] = type_to_converter(v['type'])

    return converters

and 
  
    def type_to_converter(type_name):
        type_map = {
            'Text': str,
            'Number': pd.to_numeric
        }
        return type_map[type_name]

used to `read_from_csv`

      def get_table(self, keys, value, version=None):
        [...]
        converters = make_converters(meta)
        return pd.read_csv(six.StringIO(text), converters=converters)

At first glance, it doesn't look to hard to using a importer/exporter from csv at least. 
I am still pretty wet behind the ears regarding python subclassing, so just to mention to some more seasoned pythonners
